chaoslib/litmus/platform/gke/node_memory_consumption.yml

```diff
@@ -0,0 +1,148 @@
+ ## GETTING THE PRE CHAOS NODE NAME OF APPLICATION 
+- block:
+  
+    - name: Getting the pre-chaos node name of application 
+      k8s_facts:
+        kind: Pod
+        label_selectors:
+          - "{{ a_label }}"
+      register: no_name
+
+    - set_fact:
+        node_name: "{{ no_name | json_query('resources[0].spec.nodeName')}}"
+    
+    ## PATCH THE ENGINE UUID
+    - name: Patch the engine uid
+      template:
+        src:  /chaoslib/litmus/platform/gke/node-memory-hog-daemonset.j2
+        dest: /chaoslib/litmus/platform/gke/node-memory-hog-daemonset.yml
+      vars:
+        chaosUID: "{{ chaos_uid }}"
+
+    ## CREATING A DAEMONSET FOR MEMORY-HOG
+    - name: Creating a DaemonSet for memory-hog
+      k8s:
+        state: present
+        namespace: "{{ a_ns }}"
+        src: /chaoslib/litmus/platform/gke/node-memory-hog-daemonset.yml
+      register: memory_hog_ds_result
+
+    ## CHECKING THE STATUS OF DAEMONSET
+    - name: Checking the status of DaemonSet
+      k8s_facts:
+        kind: Pod
+        namespace: "{{ a_ns }}"
+        label_selectors:
+          - app=memory-hog
+      register: pod_ds
+      until: "{{pod_ds | json_query('resources[*].status.phase') | unique | length==1}} and {{pod_ds | json_query('resources[0].status.phase') == 'Running'}}"
+      delay: 2
+      retries: 90
+
+    ## GETTING THE POD NAME OF DS FROM APPLICATION NODE
+    - name: Getting the pod name of DS from application node
+      k8s_facts:
+        kind: Pod
+        namespace: "{{ a_ns }}"
+        field_selectors:
+          - spec.nodeName={{node_name}}
+        label_selectors:
+          - app=memory-hog
+      register: po_ds_app
+
+    - set_fact:
+        pod_ds_app: "{{ po_ds_app | json_query('resources[0].metadata.name') }}"
+
+    ## WAIT FOR THE SPECIFIED RAMP TIME
+    - name: Wait for the specified ramp time before injecting chaos
+      wait_for: timeout="{{ ramp_time }}"
+      when: "ramp_time is defined and ramp_time != ''"
+
+    ## GETTING THE NODE MEMORY ALLOCATABLE
+    - name: Getting the node memory allocatable
+      k8s_facts:
+        kind: Node
+        name: "{{ node_name }}"
+      register: m_limit
+
+    - set_fact:
+        memory_limit: "{{ (m_limit | json_query('resources[0].status.allocatable.memory')).split('K')[0] | int }}"
```

Comment:
Is it compulsory that memory would be in K only?

Reply:
Yes, in `GKE` platform they use to provide the memory limit value in K only

Reply:
Can we provide a better split condition so that it will support other units?

Reply:
gcloud follows the standard unit for Memory and CPU. Like for CPU, it follows the number of cores in place of mili-cpu or other units. For memory, it follows Ki (kibi). I think that should be enough for our needs. There are a number of units present (20 units) we can't provide conversion form every unit. cc: @ksatchit

Reply:
can we make this user-defined?